File: framework/languages/dotnet/src/Zlink.Framework/Contracts/Spots/SpotRoutingContracts.cs
namespace Zlink.Framework.Contracts.Spots;

public interface IZLinkSpotRouteRefResolver
{
    ValueTask<ZLinkSpotRouteRef> ResolveSpotRouteRefAsync(
        RoutingId spotRid,
        CancellationToken cancellationToken);
}

public enum ZLinkSpotKind
{
    Invalid = 0,
    Entry = 1,
    User = 2
}

public readonly record struct ZLinkSpotRouteRef(
    string RouterChannelId,
    RoutingId TargetNodeRid,
    RoutingId SpotRid,
    ZLinkSpotKind SpotKind);

File: framework/languages/node/packages/framework/src/contracts/Spots/SpotRoutingContracts.ts
import type { RoutingId } from '../Common';

export interface ZLinkSpotRemoteAddressResolver {
  resolve(spotRid: RoutingId, signal?: AbortSignal): Promise<ZLinkSpotRemoteAddress>;
}

export enum ZLinkSpotKind {
  Invalid = 0,
  Entry = 1,
  User = 2
}

export interface ZLinkSpotRemoteAddress {
  readonly routerChannelId: string;
  readonly targetNodeRid: RoutingId;
  readonly spotRid: RoutingId;
  readonly spotKind: ZLinkSpotKind;
}
